src/Engine/ProtoCore/DSASM/HeapObjects.cs

```diff
@@ -18,11 +24,28 @@ public DSArray(int size, Heap heap)
             MetaData = new MetaData { type = (int)PrimitiveType.kTypeArray };
         }
 
-        public IEnumerable<StackValue> Values
+        /// <summary>
+        /// Create an array and populuate with input values
+        /// </summary>
+        /// <param name="values"></param>
+        /// <param name="heap"></param>
+        public DSArray(StackValue[] values, Heap heap)
+            : base(values, heap)
+        {
+            Dict = new Dictionary<StackValue, StackValue>();
```

Comment:
We can talk about this off line but it always puzzles me -- do we use `Dict.Keys` beyond basic integer-based indexing into the array, or do we do more with it? Like do we do `x = myArray["PrimaryKey"]` in DesignScript?